### ISSUE TITLE
Rare crash on Android startup caused by custom tab engagement signals

### ISSUE THREAD

**Issue by mikedawson** [UstadMobile/UstadMobile#856]
**Describe the bug**
On a small minority of devices, the Android app crashes on startup. The crash appears to be caused by Android's custom tab service engagement signals.

ACRA log:
```
java.lang.RuntimeException: java.lang.reflect.InvocationTargetException
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:503)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)
Caused by: java.lang.reflect.InvocationTargetException
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
	... 1 more
Caused by: android.os.DeadObjectException
	at android.os.BinderProxy.transactNative(Native Method)
	at android.os.BinderProxy.transact(Binder.java:1127)
	at android.support.customtabs.ICustomTabsService$Stub$Proxy.isEngagementSignalsApiAvailable(ICustomTabsService.java:512)
	at androidx.browser.customtabs.CustomTabsSession.isEngagementSignalsApiAvailable(CustomTabsSession.java:358)
	at com.ustadmobile.port.android.view.AppActivity$mCustomTabsServiceConnection$1.onCustomTabsServiceConnected(AppActivity.kt:226)
	at androidx.browser.customtabs.CustomTabsServiceConnection.onServiceConnected(CustomTabsServiceConnection.java:55)
	at android.app.LoadedApk$ServiceDispatcher.doConnected(LoadedApk.java:1730)
	at android.app.LoadedApk$ServiceDispatcher$RunConnection.run(LoadedApk.java:1762)
	at android.os.Handler.handleCallback(Handler.java:873)
	at android.os.Handler.dispatchMessage(Handler.java:99)
	at android.os.Looper.loop(Looper.java:193)
	at android.app.ActivityThread.main(ActivityThread.java:6669)
```

**To Reproduce**
1. Very difficult to reproduce.

**Expected behavior**
App starts

**Smartphone (please complete the following information):**
 - Redmi Note 8 Pro, Android 9 emulator

**Additional context**

**Comment by mhdosman** [comment]
I had a similar crash on clicking permissions on permission on course page, samsung a53,

**Comment by mhdosman** [comment]
https://github.com/UstadMobile/UstadMobile/issues/857

**Comment by mikedawson** [comment]
> I had a similar crash on clicking permissions on permission on course page, samsung a53,

As that did not happen on startup, it is a separate issue (definitely not related)

**Comment by mikedawson** [comment]
Fixed with version 0.4.90
